flow/actions/sendImageWithMessage.ts

```diff
@@ -1,29 +1,33 @@
 import {FlowCardAction} from 'homey';
 import {TelegramNotifications} from '../../app';
-import Utils from "../../utils";
-import {BL} from "betterlogiclibrary";
+import Utils from '../../utils';
+import {BL} from 'betterlogiclibrary';
 
 export default class SendImageWithMessage {
-    constructor(app: TelegramNotifications, card: FlowCardAction) {
-        card.registerRunListener(async (args) => {
-            if (app.bot != null) {
-                if (Utils.validateURL(args.url)) {
-                    await app.bot.telegram.sendPhoto(args.user.id, {
-                        filename: "",
-                        url: args.url
-                    }, {caption: await BL.decode(args.message), message_thread_id: args.user.topic})
-                        .catch(app.error);
-                } else {
-                    app.error('ERR_INVALID_PROTOCOL: Protocol "http:" not supported. Expected "https:"')
-                    throw new Error('ERR_INVALID_PROTOCOL: Protocol "http:" not supported. Expected "https:"')
-                }
-            } else {
-                app.error('Failed to start bot. Token most likely wrong.')
-                throw new Error('Failed to start bot. Token most likely wrong.')
-            }
-        });
-        card.registerArgumentAutocompleteListener(
-            'user', async (query) => Utils.userAutocomplete(app.chats, query)
-        );
-    }
+  constructor(app: TelegramNotifications, card: FlowCardAction) {
+    card.registerRunListener(async (args) => {
+      if (app.bot != null) {
+        if (Utils.validateURL(args.url)) {
+          try {
```

Comment:
Why try catch? Why not use .catch?

Reply:
This is a general code quality decision one can often find when using async/await. We could still use catch but try/catch is often more readable and helps if you want subseding code to not execute if the awaited condition failed.

`.catch()` doesn't end code flow for subsequent code. There is no subsequent code here but it's generally a good practice to keep the code style consistent.

[This article](https://www.smashingmagazine.com/2020/11/comparison-async-await-versus-then-catch/) gives a good overview about the differences of async/await and then/catch and why many people prefer one over the other.